### PR TITLE
🪟 🧹 Cleanup CatalogTree components

### PR DESCRIPTION
**Pull request by edmundito**
## What
Related to #17823 

This change decouples the SyncCatalogField from the CatalogTree so that the CatalogTree component can be more expandable and maintainable and separate from the ConnectionForm

## How
Update the `SyncCatalogField`to only be responsible for rendering the field itself. Leaves the rest to the `CatalogTree` now, including heading, search, and loading state. All of these components have been moved to the CatalogTree section.

Updates most files moved to use SCSS over styled-components.

## Recommended reading order
1. `CatalogTree.tsx`
2. `SyncCatalogField.tsx`
3. Rest of code

### CODE REVIEW

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeRows.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import { CatalogSection } from "./CatalogSection";
+import styles from "./CatalogTreeRows.module.scss";
+
+interface CatalogTreeRowsProps {
+  streams: SyncSchemaStream[];
+  onStreamChanged: (stream: SyncSchemaStream) => void;
+}
+
+export const CatalogTreeRows: React.FC<CatalogTreeRowsProps> = ({ streams, onStreamChanged }) => {
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeRows.tsx:17]: This is is what CatalogTree used to be. Now CatalogTree contains all the headers, bulk edit, and search bar.

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+%catalogHeader {
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss:1]: This is repeated between the CatalogTreeHeader and the CatalogTreeSubheader module. I'm not sure how to reuse this with SCSS modules.

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { FormattedMessage } from "react-intl";
+import styled from "styled-components";
+
+import { Cell, Header } from "components/SimpleTableComponents";
+
+import { useConnectionFormService } from "hooks/services/ConnectionForm/ConnectionFormService";
+
+import styles from "./CatalogTreeSubheader.module.scss";
+
+const SubtitleCell = styled(Cell).attrs(() => ({ lighter: true }))`
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.tsx:12]: Cell is still based on a styled-component so I left this as-is.

File: airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionFormFields.tsx [hunk 4]
```diff
@@ -55,7 +55,7 @@ export const ConnectionFormFields: React.FC<ConnectionFormFieldsProps> = ({ valu
           <ScheduleField />
         </Section>
         <Section>
-          <Text as="h5">
+          <Text as="h2" size="sm">
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionFormFields.tsx:58]: This looked larger than anything the other headings, so I fixed it.

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx [hunk 5]
```diff
@@ -1,59 +1,65 @@
...
-              updateStream={onUpdateStream}
-              changedSelected={changedStreams.includes(streamNode) && mode === "edit"}
-            />
-          )}
-        </Field>
-      ))}
-    </>
+    <BatchEditProvider nodes={streams} update={onStreamsChanged}>
+      <LoadingBackdrop loading={isLoading}>
+        {mode !== "readonly" && <CatalogTreeSearch onSearch={setSearchString} />}
```
Comment (tealjulia) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx:55]: why shouldn't users search the catalog for a disabled connection?

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss (hunk 2 above)
Comment (tealjulia) [reply airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss:1]: It looks like mixins may be favored over placeholders for modules: https://github.com/sass/dart-sass/issues/1042

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  min-height: 33px;
+  margin-bottom: 0 !important;
+  margin-left: 20px;
```
Comment (tealjulia) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss:4]: can use variable

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.catalogSubheader {
+  @extend %catalogHeader;
+
+  padding-top: 10px;
```
Comment (tealjulia) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss:10]: can use variable

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  margin-left: 155px;
+}
+
+.readonlyCatalogSubheader {
+  @extend %catalogHeader;
+
+  padding-top: 10px;
```
Comment (tealjulia) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTreeSubheader.module.scss:17]: can use variable

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx [hunk 9]
```diff
@@ -1,59 +1,65 @@
-import { Field, FieldProps, setIn, useFormikContext } from "formik";
-import React, { useCallback } from "react";
+import React, { useCallback, useMemo, useState } from "react";
+
+import { LoadingBackdrop } from "components/ui/LoadingBackdrop";
 
 import { SyncSchemaStream } from "core/domain/catalog";
-import { AirbyteStreamConfiguration } from "core/request/AirbyteClient";
+import { BatchEditProvider } from "hooks/services/BulkEdit/BulkEditService";
```
Comment (tealjulia) [inline airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx:6]: It feels like we're using `Batch` and `Bulk` interchangeably in this service.  Maybe not a fix for this PR though.

**Review by teallarson — APPROVED** [review]
Tested out creating and editing connections and did not find any regressions.

Left a few comments mostly RE: sass variables, though if these components are going to be overhauled in the design update I don't have strong feelings about using them.

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx (hunk 5 above)
Comment (edmundito) [reply airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx:55]: I'm not sure why this was wrapped under readonly in the first place. Maybe it's not needed?

Comment (tealjulia) [reply airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx:55]: It's entirely possible I was a bit too overzealous in shutting down every `input` element when I implemented the "readonly" mode for deleted connections.

File: airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx (hunk 9 above)
Comment (edmundito) [reply airbyte-webapp/src/views/Connection/CatalogTree/CatalogTree.tsx]: As I was working on this more, I noticed this as well. Updated!

**Comment by edmundito** [comment]
@tealjulia Agreed. I would consider parts of the component such as `CatalogTreeSubheader` to be legacy, so I will leave the CSS as it is.

**Pull request merged**